cosmrs/src/base.rs

```diff
@@ -136,6 +136,16 @@ pub struct Coin {
     pub amount: u128,
 }
 
+impl Coin {
+    /// Constructor
+    pub fn new(amount: u128, denom: &str) -> Result<Self> {
+        Ok(Coin {
+            amount,
+            denom: Denom::from_str(denom)?,
```

Comment:
Slight simplification:

```suggestion
            denom: denom.parse()?,
```